packages/runtime/client/rest/src/index.ts

```diff
@@ -100,7 +104,15 @@ export interface RestAPI {
     contractId: ContractId,
     range?: TransactionsRange
   ): Promise<Transactions.GetTransactionsForContractResponse>;
-  //   createTransactionForContract: Transactions.POST; // - https://docs.marlowe.iohk.io/api/create-transactions // TODO: Jamie, lets unify names
+  /**
+   * Create an unsigned transaction which applies inputs to a contract.
+   * @see {@link https://docs.marlowe.iohk.io/api/apply-inputs-to-contract}
+   */
+  createTransactionForContract(
+    contractId: ContractId,
+    postTransactionsRequest: Transactions.PostTransactionsRequest,
+    addressesAndCollaterals: AddressesAndCollaterals
```

Comment:
To be consistent with other endpoints, I think this should receive a single object with clear markings on optional and required fields. Use [this](https://github.com/input-output-hk/marlowe-ts-sdk/blob/1af1b33d8a0b2047e80563e077e288f6511ae898/packages/runtime/client/rest/src/contract/endpoints/collection.ts#L203) as a starting point, and notice [the changes](https://github.com/input-output-hk/marlowe-ts-sdk/pull/42/files#diff-a61d1081ab5a7fac0cc6b84a9e54a67226c806c18fe58e2f308f40972e9bcaedR213-R239) here from my last unmerged PR that is being reviewed

For example, in addressAndCollateral only the collateral is mandatory, usedAddresses and collateralUTxOs arent.

Also, the response should probably be wrapped in its own Interface, notice to improve legibility.

Notice the exported documentation

<img width="980" alt="Screenshot 2023-10-18 at 10 34 28" src="https://github.com/input-output-hk/marlowe-ts-sdk/assets/2634059/e9165578-4a2e-42d9-9f5d-97dd04dde4d1">

Reply:
Also, following the 1-1 mapping, the name of this endpoint should be `applyInputsToContract`.
I think the previous name (the one on the ticket) was from Jamies straw suggestion

Reply:
Alright, no problems. I saw that there was a discrepancy between our names and the names in the [Runtime API](https://docs.marlowe.iohk.io/api). I will fix this.